src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  const womenWithChildren = withChildren
+    ? women.filter(woman => people.some(person => woman.name === person.mother))
+    : women;
+
+  const averageAge = womenWithChildren.reduce((sum, woman) =>
+    sum + (woman.died - woman.born), 0) / womenWithChildren.length || 0;
```

Comment:
Create separate function `calculateAverageAge` and move this logic to this function.
Now you have the logic with reduce repeated in all 3 functions.
Make it so that the function takes an array of numbers and the length by which it needs to be divided